biothings/hub/databuild/builder.py

```diff
@@ -1271,10 +1271,17 @@ def archive_merge(self, merge_name):
                 "No metadata found for merged collection '%s'" % merge_name)
         self.delete_merged_data(merge_name)
 
-    def get_query_for_list_merge(self, only_archived):
+    def get_query_for_list_merge(self, only_archived, status=None):
         q = {"archived": {"$exists": 0}}
         if only_archived:
             q = {"archived": {"$exists": 1}}
+        if status:
+            if status == "success":
+                q["jobs.status"] = {
+                    "$not": {"$in": ["failed", "canceled"]}
+                }
+            if status == "failed":
```

Comment:
@sengineer0 if we have a status value of "canceled", we should support filtering on it too, right?

Reply:
right. I will add "canceled" status soon

Reply:
@newgene it's updated

Reply:
@sengineer0 that works, but I kind of don't like the parallel `if` blocks here. using `elif` and `else` instead here? and you can probably merge both "failed" and "canceled" cases into just one code block I think? Anyway, just need a bit of optimization here.

Reply:
@newgene  let's me see

Reply:
@newgene  it's updated